src/test/unittest/unittest.h

```diff
@@ -469,4 +469,41 @@ int ut_pthread_join(const char *file, int line, const char *func,
 #define	PTHREAD_JOIN(thread, value_ptr)\
     ut_pthread_join(__FILE__, __LINE__, __func__, thread, value_ptr)
 
+/*
+ * mocks...
+ */
+#define	FUNC_REAL_DECL(name, ret_type, ...)\
```

Comment:
Looks pretty good!

I only have a couple of suggestions.
Regarding the macro names:
1) The first two macros are internal, right? So, what about "_" prefix? 
2) FUNC_MOCK_RUN_DEFAULT => FUNC_MOCK_DEFAULT?
3) FUNC_WILL_RETURN => FUNC_MOCK_RET_ALWAYS or FUNC_MOCK_ALWAYS_RET?

Also, if I understand it well, `FUNC_WILL_RETURN(foo, x)` in practice is equivalent to:
```C
FUNC_MOCK(foo, typeof (x)) { FUNC_MOCK_RUN_RET_DEFAULT(x); }
```
So, there's not a big saving... But maybe it's worth to have such macro if it's used frequently.
I would only prefer to define it as:
```C
#define	FUNC_WILL_RETURN(name, type_ret, ret)\
	(type_ret) __wrap_##name(void) {\
		return (ret); };
```